makroer/sykehusopphold.sas

```diff
@@ -0,0 +1,354 @@
+/* This is a macro to replace the old Episode of Care */
+/* It takes the aggrshoppID_Lnr from NPR as a starting point, 
+   and adds on overf�ring to other sykehus within a specified time (defult 8 hours) */
+   
+%macro sykehusopphold(
+  dsn=, 
+  overforing_tid=28800 /*tid mellom opphold sammensl�ing*/, 
+  forste_hastegrad = 1, 
+  behold_datotid=0,
+  nulle_liggedogn = 0,
+  minaar=0);
+/*!
+
+ ### Beskrivelse
+Makro for � markere opphold som regnes som en sykehusepisode. 
+Inndatasettet m� inneholde pid inndato utdato inntid og uttid
+
+```
+%sykehusopphold(dsn=, overforing_tid=28800, forste_hastegrad=1, behold_datotid=0, nulle_liggedogn=0, minaar=0);
+```
+### Parametre
+
+### Input
+- `dsn`: datasett man utf�rer analysen p� 
+- `overforing_tid` (=28800): tidskrav i sekunder (default 8 timer) for opphold med tidsdifferanse - m� v�re et tall
+- `forste_hastegrad` (=1): Hastegrad for f�rste kontakt hvis ulik 0. Hvis 0: hastegrad for f�rste d�gnopphold
+- `behold_datotid` (=0): Hvis ulik 0, s� beholdes disse 
+- `nulle_liggedogn` (=0): Hvis forskjellig fra null, s� settes antall SHO_liggedogn til null hvis opphold er <8 timer
+- `minaar` (=0): dersom man �nsker � bruke f�rste opphold for � definere SHO_Aar
+  
+Episode of care omfatter da:
+- Dersom inndatotid p� nytt opphold er f�r utdatotid p� forrige opphold
+- Dersom nytt opphold er innen *x* sekunder fra utdatotid p� forrige opphold
+
+
+ 
+### Output
+Makroen lager 12 (eventuelt 14) nye variabler:
+1.  `SHO_nr_pid`: Nummererte SHO-opphold pr pid, dersom SHO best�r av flere opphold har disse samme nummer
+2.  `SHO_id`: Unik ID per sykehusopphold (pid+SHO_nr_pid)
+3.  `SHO_Intern_nr`: Nummerer oppholdene innenfor hver SHO for hver pid
+4.  `SHO_Antall_i_SHO`: Antall opphold i SHO
+5.  `SHO_inndato`: inndato p� f�rste opphold i SHO
+6.  `SHO_utdato`: utdato p� siste opphold i SHO
+7.  [eventuelt] `SHO_inndatotid`: inntid (dato og tidspunkt) p� f�rste opphold i SHO (hvis behold_datotid ulik 0))
+8.  [eventuelt] `SHO_utdatotid`: uttid (dato og tidspunkt) p� siste opphold i SHO (hvis behold_datotid ulik 0))
+9.  `SHO_aar`: �r ved utskriving - alternativt minaar=1 (�r ved f�rste opphold)
+10. `SHO_liggetid`: tidsdifferanse mellom inndatotid p� det f�rste oppholdet og utdatotid p� det siste oppholdet i SHO
+11. `SHO_aktivitetskategori3`: 1 hvis ett av oppholdene er d�gn, eller 2 hvis ett av oppholdene er dag (og ingen d�gn), eller 3 hvis oppholdene er kun poli
+12. `SHO_hastegrad`: 1 hvis ett av oppholdene er akutt, 4 ellers
+13. `SHO_forste_hastegrad`: Hastegrad for f�rste avdelingsopphold. 1 for akutt, 4 for elektiv
+14. `SHO_uttilstand`: max(uttilstand) for oppholdene i en SHO ("som d�d" hvis ett av oppholdene er "som d�d")
+
+
+### Endringslogg
+- 07.09.2021 Opprettet av Janice Shu
+ */
+
```

Comment:
Burde det være et drop-statement først? Slik at hvis en kjører makroen to ganger så fjernes de gamle sho-variablene først? 
Hvis ikke så kommer det feilmelding ved andre gangs kjøring.